neo/utils.py

```diff
@@ -10,6 +10,86 @@
 import quantities as pq
 
 
+def merge_anasiglist(anasiglist, axis=1):
+    """
+    Merges neo.AnalogSignal objects into a single object.
+
+    Units, sampling_rate, t_start, t_stop and signals shape must be the same
+    for all signals. Otherwise a ValueError is raised.
+
+    Parameters
+    ----------
+    anasiglist: list of neo.AnalogSignal
+        list of analogsignals that will be merged
+    axis: int
+        axis along which to perform the merging
+        `axis = 1` corresponds to stacking the analogsignals
```

Comment:
By implementing the combination of signals along two different axis you are covering two different scenarios, that require different checks on the neo level. Eg. when stacking signals (ie merging multiple channels in one Analogsignal) it is useful to ensure they have the same length and cover the same time range. However for concatenating analogsignals along time, these conditions do not need to be fulfilled. I would argue therefore these different cases should also be implemented in separate functions. This would also make it easier for the users to decide which of the actions they actually want to use.

Reply:
Another feature that needs to be treated differently depending on the axis are the array_annotations. I am not sure in how far this implementation is taking care of this, but I think we should double check here.